File: backend/Python/app/core/result.py
from typing import Generic, TypeVar, Union

T = TypeVar("T")

class Ok(Generic[T]):
    def __init__(self, value: T):
        self.ok = True
        self.value = value

class Err:
    def __init__(self, code: str, message: str):
        self.ok = False
        self.code = code
        self.message = message

Result = Union[Ok[T], Err]

def ok(value: T) -> Result:
    return Ok(value)

def err(code: str, message: str) -> Result:
    return Err(code, message)

File: backend/Node/src/core/result.ts
export type Result<T> = Success<T> | Failure;

export interface Success<T> {
	ok: true;
	value: T;
}

export interface Failure {
	ok: false;
	code: string;
	message: string;
}

export const ok = <T>(value: T): Success<T> => ({ ok: true, value });
export const fail = (code: string, message: string): Failure => ({ ok: false, code, message });
